src/python/gudhi/representations/vector_methods.py

```diff
@@ -489,7 +503,10 @@ def fit(self, X, y=None):
         """
         if self.mode == "vector":
             _grid_from_sample_range(self, X)
-            self.step_ = self.grid_[1] - self.grid_[0]
+            if sum([len(x) for x in X]) > 0:
```

Comment:
Maybe `if self.sample_range_fixed_[0] != self.sample_range_fixed_[1]` as in the other class? Or always compute `self.grid_[1] - self.grid_[0]`, and if it is NaN, replace with 0?

Otherwise I am afraid we are going to break the case where the user manually specified a valid range and then fits on an empty list since there is nothing to learn.